### PR TITLE
Add Edit and Destroy to Iterations and Engagements

### PR DESCRIPTION
**Pull request by julies642**
Allow all Iterations and Engagements to be updated or deleted to keep everything organized.

### CODE REVIEW

File: features/edit_destroy.feature [hunk 1]
```diff
@@ -26,38 +22,79 @@ Background: Logged in
...
+  Then I should see "2017-03-25" has button "Edit"
+  And I should see "2017-08-25" has button "Edit"
+  And I should see "2017-03-25" has button "Destroy"
+  And I should see "2017-08-25" has button "Destroy"
+
+Scenario: Editing existing engagements is successful
+  #Story ID: #152689950
+  And I press "Edit" for "2017-03-25"
+  Then I should see "Editing Engagement for"
+  And I press "Save"
```
Comment (kimhoang73) [inline features/edit_destroy.feature:48]: I'm not sure if this was tested in other cucumber tests, but maybe we can add actual steps to edit the engagement, then check to see of those changes were actually saved

File: config/routes.rb [hunk 2]
```diff
@@ -8,11 +8,13 @@
   # route /apps/:app_id/engagements/:engagement_id
   resources :apps do
     resources :engagements, :except => :index
+    resources :engagements, :only => :destroy, :as => 'engagement_destroy'
```
Comment (csungwon) [inline config/routes.rb]: Was there any reason why you changed the path names? Seems like we already have :app_engagement_path and :engagement_iteration_path

Comment (kyriakipapadimitriou) [reply config/routes.rb]: There are indeed existing paths but they did not have a 'prefix' like edit, create, index, etc. when I ran "rake routes". I just created a prefix with the same path to be more readable in the link_to call.

Comment (csungwon) [reply config/routes.rb]: oh, they all share `app_engagement`. Just use it with different HTTP method.
### Example ###
* app_comment + get => comments#show
* app_comment + delete => comments#destroy

**Review by kimhoang73 — APPROVED** [review]
looks good :D !

File: features/edit_destroy.feature (hunk 1 above)
Comment (kyriakipapadimitriou) [reply features/edit_destroy.feature:48]: That's a good suggestion. I initially had steps editing the engagement to see if it is working but then saw engagement_users.feauture already does that so I deleted them. Same applies for editing an iteration.

File: config/routes.rb (hunk 2 above)
Comment (kyriakipapadimitriou) [reply config/routes.rb]: Ohh yes I see. There was no need to create a prefix. I just changed the method and it worked. Thanks Sean! Pushing the changes now

**Review by csungwon — APPROVED** [review]
Looks good to me! Well done.

**Pull request merged**